src/mesh.jl

```diff
@@ -143,21 +143,21 @@ end
 
 A grid embedded in a `Dim`-dimensional space with given coordinate reference system `CRS`.
 """
-const Grid{Dim,CRS} = Mesh{Dim,CRS,GridTopology{Dim}}
+const Grid{CRS,Dim} = Mesh{CRS,GridTopology{Dim}}
 
 """
     SubGrid{Dim,CRS}
 
 A view of a grid in a `Dim`-dimensinoal space with given coordinate reference system `CRS`.
 """
-const SubGrid{Dim,CRS} = SubDomain{Dim,CRS,<:Grid{Dim,CRS}}
+const SubGrid{CRS,Dim} = SubDomain{CRS,<:Grid{CRS,Dim}}
 
 """
     vertex(grid, ijk)
 
 Convert Cartesian index `ijk` to vertex on `grid`.
 """
-vertex(g::Grid{Dim}, ijk::CartesianIndex{Dim}) where {Dim} = vertex(g, ijk.I)
+vertex(g::Grid{CRS,Dim}, ijk::CartesianIndex{Dim}) where {CRS,Dim} = vertex(g, ijk.I)
```

Comment:
let's omit CRS and Dim here, I don't think we are gaining much by restricting ijk